q2_sample_classifier/visuals.py

```diff
@@ -102,8 +102,27 @@ def _linear_regress(actual, pred):
 def _plot_heatmap_from_confusion_matrix(cm, palette, vmin=None, vmax=None):
     palette = _custom_palettes()[palette]
     plt.figure()
-    return sns.heatmap(cm, vmin=vmin, vmax=vmax, cmap=palette,
-                       cbar_kws={'label': 'Proportion'})
+    scaletron, labelsize, dpi = 20, 8, 100
+    sns.set(rc={'xtick.labelsize': labelsize, 'ytick.labelsize': labelsize,
+            'figure.dpi': dpi})
+    fig, (ax, cax) = plt.subplots(ncols=2, constrained_layout=True)
+    heatmap = sns.heatmap(cm, vmin=vmin, vmax=vmax, cmap=palette, ax=ax,
+                          cbar_ax=cax, cbar_kws={'label': 'Proportion'},
+                          square=True, xticklabels=True, yticklabels=True)
+
+    # Resize the plot dynamically based on number of classes
+    hm_pos = ax.get_position()
+    scale = len(cm) / scaletron
+    # set minimum color bar height for readability
+    cbar_height = scale if (scale >= .15) else .15
```

Comment:
Please factor out the 0.15 as a local constant, and add to line 105 above (the scaler, labelsize, dpi line).

Reply:
```suggestion
    cbar_height = scale if (scale > .15) else .15
```

Dropping the `=` has the same effect, and is one less thing to mentally parse when reading.

Reply:
`max(scale, 0.15)` might be clearer.